### ISSUE TITLE
Clicking on the window title for the first time moves the mouse to the left corner

### ISSUE THREAD

**Issue by ellipticaldoor** [libsdl-org/SDL#5332]
The first time I open an SDL2 window, after I click on the window title, the cursor gets moved to the left corner.

Here is a video showing this

https://user-images.githubusercontent.com/3449006/152881394-ed511ed2-ac1d-4f75-9f03-82a2da18eea5.mp4

I'm on macos monterey 12.1 and this can be reproduced with this simple placeholder sdl2 app:

```c++
#include <SDL2/SDL.h>
#include <stdio.h>

#define SCREEN_WIDTH 640
#define SCREEN_HEIGHT 480

int main(int argc, char *args[])
{
  SDL_Window *window = NULL;
  SDL_Surface *screenSurface = NULL;
  if (SDL_Init(SDL_INIT_VIDEO) < 0)
  {
    fprintf(stderr, "could not initialize sdl2: %s\n", SDL_GetError());
    return 1;
  }
  window = SDL_CreateWindow(
    "hello_sdl2",
    SDL_WINDOWPOS_UNDEFINED,
    SDL_WINDOWPOS_UNDEFINED,
    SCREEN_WIDTH,
    SCREEN_HEIGHT,
    SDL_WINDOW_SHOWN | SDL_WINDOW_RESIZABLE);
  if (window == NULL)
  {
    fprintf(stderr, "could not create window: %s\n", SDL_GetError());
    return 1;
  }
  screenSurface = SDL_GetWindowSurface(window);
  SDL_FillRect(
    screenSurface, NULL, SDL_MapRGB(screenSurface->format, 0xFF, 0xFF, 0xFF));
  SDL_UpdateWindowSurface(window);
  SDL_Event event;
  int gogogo = 1;
  while (gogogo)
  {
    SDL_WaitEvent(&event);
    if (event.type == SDL_QUIT) gogogo = 0;
  }
  SDL_DestroyWindow(window);
  SDL_Quit();
  return 0;
}

```

**Comment by captnjameskirk** [comment]
Just wanted to note that I've also encountered this issue. It always occurs on the first click in the titlebar when the window is resizable. I went so far as to try making the window initially not resizable, immediately simulating a click in the titlebar and then setting the window as resizable. The issue still occurs on the first click in the titlebar after the window is resizable.

The only way I was able to get around it was to leave the window not resizable and use a HitTestCallback to manually implement dragging/resizing (extremely hacky and ugly, and it loses the native "resizing" cursor handles).

I tested on a few releases go back to 2.0.12 with OpenGL and Metal on Intel and M1 hardware, but only with Big Sur and Monterey (didn't have older macOS versions to test further). Seems to have been around a while.

The good news is I can't reproduce with 2.0.20. but I'm not sure if it was intentionally fixed or just a happy accident since it's not mentioned in the release notes.

If it wasn't intentionally fixed, the bug in previous releases should probably still be tracked down so it doesn't get unintentionally reintroduced again.
